onnxruntime/core/session/abi_session_options.cc

```diff
@@ -177,9 +178,43 @@ ORT_API_STATUS_IMPL(OrtApis::AddSessionConfigEntry, _Inout_ OrtSessionOptions* o
 
 ORT_API_STATUS_IMPL(OrtApis::AddInitializer, _Inout_ OrtSessionOptions* options, _In_z_ const char* name,
                     _In_ const OrtValue* val) {
+  API_IMPL_BEGIN
   auto st = options->value.AddInitializer(name, val);
   if (!st.IsOK()) {
     return onnxruntime::ToOrtStatus(st);
   }
   return nullptr;
-}
\ No newline at end of file
+  API_IMPL_END
+}
+
+ORT_API_STATUS_IMPL(OrtApis::AddExternalInitializers, _In_ OrtSessionOptions* options,
+                    _In_reads_(input_len) const char* const* input_names,
+                    _In_reads_(input_len) const OrtValue* const* inputs, size_t input_len) {
+
+#if !defined(ORT_MINIMAL_BUILD) && !defined(DISABLE_EXTERNAL_INITIALIZERS)
+  API_IMPL_BEGIN
+  onnxruntime::InlinedVector<std::string> names;
+  onnxruntime::InlinedVector<OrtValue> values;
+  names.reserve(input_len);
+  values.reserve(input_len);
+  for (size_t i = 0; i < input_len; ++i) {
+    names.emplace_back(input_names[i]);
+    values.emplace_back(*inputs[i]);
```

Comment:
>inputs [](http://example.com/codeflow?start=25&length=6)

do we need to check that inputs[i] != nullptr?

Reply:
Good idea although there is not a check against garbage.